Library/Homebrew/keg_relocate.rb

```diff
@@ -69,8 +74,9 @@ def replace_text_in_files(relocation, files: nil)
       s = first.open("rb", &:read)
 
       replacements = {
-        relocation.old_prefix => relocation.new_prefix,
-        relocation.old_cellar => relocation.new_cellar,
+        relocation.old_prefix  => relocation.new_prefix,
+        relocation.old_cellar  => relocation.new_cellar,
+        relocation.old_library => relocation.new_library,
```

Comment:
This logic is only required `if HOMEBREW_PREFIX == HOMEBREW_REPOSITORY` so should be conditionally added  (in the other branch to how `relocation.old_repository` is conditionally added below).

Reply:
> It's currently unlikely but we need to ensure that we do not have non-Library references to `HOMEBREW_REPOSITORY for the above statement to be true.

I've added an audit to ensure that any references to `HOMEBREW_REPOSITORY` are within `HOMEBREW_LIBRARY`.